data-mapper-logstash/swarm.sh

```diff
@@ -1,82 +1,131 @@
   for file in "${COMPOSE_FILE_PATH}"/pipeline/*.conf; do
     sed -i "s/\$ES_HOSTS/${ES_HOSTS}/g" "${file}"
   done
 }
 
-if [[ "$MODE" == "dev" ]]; then
-  log info "Running Data Mapper Logstash package in DEV mode"
-  logstash_dev_compose_param="-c ${COMPOSE_FILE_PATH}/docker-compose.dev.yml"
-else
-  log info "Running Data Mapper Logstash package in PROD mode"
-  logstash_dev_compose_param=""
-fi
-
-if [[ "$LOGSTASH_DEV_MOUNT" == "true" ]]; then
-  if [[ -z $LOGSTASH_PACKAGE_PATH ]]; then
-    log error "LOGSTASH_PACKAGE_PATH environment variable not specified. Please specify LOGSTASH_PACKAGE_PATH as stated in the README."
-    exit 1
-  fi
-
-  log info "Running Data Mapper Logstash package with dev mount"
-  logstash_dev_mount_compose_param="-c ${COMPOSE_FILE_PATH}/docker-compose.dev-mnt.yml"
-else
-  logstash_dev_mount_compose_param=""
-fi
-
-if [[ "${ACTION}" == "init" ]] || [[ "${ACTION}" == "up" ]]; then
+function initialize_package() {
+  local logstash_dev_compose_filename=""
+  local logstash_dev_mount_compose_filename=""
+  local logstash_temp_compose_filename=""
 
-  if [[ "${STATEFUL_NODES}" == "cluster" ]]; then
-    export LOGSTASH_YML_CONFIG="logstash-logstash.cluster.yml"
+  if [[ "$MODE" == "dev" ]]; then
+    log info "Running Data Mapper Logstash package in DEV mode"
+    logstash_dev_compose_filename="docker-compose.dev.yml"
   else
-    export LOGSTASH_YML_CONFIG="logstash-logstash.yml"
+    log info "Running Data Mapper Logstash package in PROD mode"
   fi
 
-  inject_pipeline_elastic_hosts
+  if [[ "$LOGSTASH_DEV_MOUNT" == "true" ]]; then
+    log warn "LOGSTASH_DEV_MOUNT is enabled: Please make sure TO REPLACE ES_HOSTS MANUALLY IN ALL THE FILES inside pipeline folder!"
+  else
+    inject_pipeline_elastic_hosts
 
-  config::set_config_digests "${COMPOSE_FILE_PATH}"/docker-compose.yml
+    config::generate_service_configs data-mapper-logstash /usr/share/logstash "${COMPOSE_FILE_PATH}/pipeline" "${COMPOSE_FILE_PATH}" "logstash"
+    logstash_temp_compose_filename="docker-compose.tmp.yml"
+  fi
 
-  config::generate_service_configs data-mapper-logstash /usr/share/logstash "${COMPOSE_FILE_PATH}/pipeline" "${COMPOSE_FILE_PATH}" logstash
-  logstash_temp_compose_param="-c ${COMPOSE_FILE_PATH}/docker-compose.tmp.yml"
+  (
+    dev_mount_logstash
 
-  try "docker stack deploy -c ${COMPOSE_FILE_PATH}/docker-compose.yml $logstash_dev_compose_param $logstash_dev_mount_compose_param $logstash_temp_compose_param instant" "Failed to deploy Data Mapper Logstash"
+    docker::deploy_service "${COMPOSE_FILE_PATH}" "docker-compose.yml" "$logstash_dev_compose_filename" "$logstash_dev_mount_compose_filename" "$logstash_temp_compose_filename"
+    docker::deploy_sanity "${service_name}"
+  ) || {
+    log error "Failed to deploy Data Mapper Logstash package"
+    exit 1
+  }
+}
 
-  docker::await_container_startup data-mapper-logstash
-  docker::await_container_status data-mapper-logstash Running
+function scale_services_down() {
```

Comment:
This should be `scale_service_down` if it only scales one service down.

Also, I'm guessing this is going to be in quite a few subsequent PRs, so maybe we should make this into a utility?

Reply:
Yes I refactored this in the latest two PRs